api/src/main/java/org/pmiops/workbench/workspaceadmin/WorkspaceAdminServiceImpl.java

```diff
@@ -354,6 +361,67 @@ public void setAdminUnlockedState(String workspaceNamespace) {
     workspaceDao.save(dbWorkspace.setAdminLocked(false));
     adminAuditor.fireUnlockWorkspaceAction(dbWorkspace.getWorkspaceId());
   }
+
+  @Override
+  public void setResearchPurposeApproved(
+      String workspaceNamespace, String firecloudName, boolean approved) {
+    DbWorkspace workspace = workspaceDao.getRequired(workspaceNamespace, firecloudName);
+    if (workspace.getReviewRequested() == null || !workspace.getReviewRequested()) {
+      throw new BadRequestException(
+          String.format(
+              "No review requested for workspace %s/%s.", workspaceNamespace, firecloudName));
+    }
+
+    Boolean existingApproval = workspace.getApproved();
+    if (existingApproval != null) {
+      throw new BadRequestException(
+          String.format(
+              "DbWorkspace %s/%s already %s.",
+              workspaceNamespace, firecloudName, existingApproval ? "approved" : "rejected"));
+    }
+    workspace.setApproved(approved);
+    workspaceDao.saveWithLastModified(workspace);
+
+    // RW-7087 replace with a new workspaceAuditor action (fireReviewAction?)
+    // because this uses the deprecated DbAdminActionHistory
+    userService.logAdminWorkspaceAction(
+        workspace.getWorkspaceId(),
+        "research purpose approval",
+        workspace.getApproved(),
+        existingApproval);
+  }
+
+  @Override
+  public List<Workspace> getWorkspacesForReview() {
+    return workspaceDao.findByApprovedIsNullAndReviewRequestedTrueOrderByTimeRequested().stream()
+        .map(workspaceMapper::toApiWorkspace)
+        .collect(Collectors.toList());
+  }
+
+  @Override
+  public DbWorkspace setPublished(
```

Comment:
FYI we are in the process of modifying this method: #6005

Reply:
Thanks, this is not urgent (and I'm still fixing tests), so definitely proceed with that first

Reply:
Looks like this doesn't interact with the newer version of the PR: https://github.com/all-of-us/workbench/pull/6027 , so going to proceed.